src/main/java/online/books/webstore/repository/book/AuthorSpecificationProvider.java

```diff
@@ -0,0 +1,23 @@
+package online.books.webstore.repository.book;
+
+import java.util.Arrays;
+import online.books.webstore.model.Book;
+import online.books.webstore.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
+    private final String key = "author";
```

Comment:
What about constant? I mean every class what consist key

Reply:
What do you mean?  key should be public static final ?